public/views/Search.vue

```diff
@@ -1,110 +1,287 @@
 import { SEARCH_ROUTE, JOIN_DATASETS_ROUTE } from "../store/route/index";
+import { actions as viewActions } from "../store/view/module";
+import { createRouteEntry, overlayRouteEntry } from "../util/routes";
+import { spinnerHTML } from "../util/spinner";
 
 export default Vue.extend({
   name: "search-view",
 
   components: {
-    SearchBar,
-    DatasetSearchResults,
-    ModelSearchResults,
-    DatasetPreviewCard,
+    DatasetPreview,
     FileUploader,
-    FileUploaderStatus
+    FileUploaderStatus,
+    ModelPreview,
+    SearchBar
   },
 
   data() {
     return {
       isPending: false,
+      sorting: {
+        asc: true,
+        type: "name"
+      },
+      tab: "datasets",
       uploadData: {},
       uploadStatus: ""
     };
   },
 
   computed: {
-    terms(): string {
-      return routeGetters.getRouteTerms(this.$store);
+    filteredDatasets(): Dataset[] {
+      return datasetGetters.getFilteredDatasets(this.$store);
+    },
+
+    filteredDatasetsIds(): Set<String> {
+      const ids = this.filteredDatasets.map(dataset => dataset.id);
+      return new Set(ids);
+    },
+
+    filteredModels(): Model[] {
+      const models = modelGetters.getModels(this.$store);
+
+      // Only display the models using dataset that the search bar has found.
+      return models.filter(model =>
+        this.filteredDatasetsIds.has(model.datasetId)
+      );
     },
 
     nbSearchDatasets(): number {
-      return datasetGetters.getCountOfFilteredDatasets(this.$store);
+      return this.filteredDatasets.length ?? 0;
     },
+
     nbSearchModels(): number {
-      return modelGetters.getCountOfModels(this.$store);
+      return this.filteredModels.length ?? 0;
+    },
+
+    /**
+     * List of search results to be displayed.
+     */
+    searchResults(): any[] {
+      const results = [] as any[];
+
+      // If tab is either 'models' or 'all' we display the models.
+      if (this.tab !== "datasets") {
+        const models = this.filteredModels.map(model => {
+          return { type: "model", model };
+        });
+        results.push(...models);
+      }
+
+      // If tab is either 'datasets' or 'all' we display the datasets.
+      if (this.tab !== "models") {
+        const datasets = this.filteredDatasets.map(dataset => {
+          return { type: "dataset", dataset };
+        });
+        results.push(...datasets);
+      }
+
+      return results;
+    },
+
+    /**
+     * Sort the results based on the sorting selected.
+     */
+    sortedResults(): any[] {
+      return this.searchResults.slice().sort((a, b) => {
+        // Sort by recent activity
+        // if (this.sorting.type === "recent") {
+        // ...
+        // }
+
+        // Sort by name
+        if (this.sorting.type === "name") {
+          a = this.getNameFromResult(a);
+          b = this.getNameFromResult(b);
+          return this.sorting.asc ? a.localeCompare(b) : b.localeCompare(a);
```

Comment:
[`localCompare()` is well supported.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare)